bees/discordbee/discordbee.go

```diff
@@ -0,0 +1,143 @@
+/*
+ *    Copyright (C) 2019      CalmBit
+ *                  2014-2019 Christian Muehlhaeuser
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU Affero General Public License as published
+ *    by the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Authors:
+ *      CalmBit <calmbit@posto.net>
+ *      Christian Muehlhaeuser <muesli@gmail.com>
+ */
+
+// Package discordbee is
+package discordbee
+
+import (
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/muesli/beehive/bees"
+)
+
+// DiscordBee is
+type DiscordBee struct {
+	bees.Bee
+
+	apiToken      string
+	triggerPhrase string
+	eventChan     chan bees.Event
+	discord       *discordgo.Session
+}
+
+// Run executes the Bee's event loop.
+func (mod *DiscordBee) Run(eventChan chan bees.Event) {
+	mod.eventChan = eventChan
+
+	mod.Logf("Starting discord bee with apiToken %s", mod.apiToken)
+	d, err := discordgo.New("Bot " + mod.apiToken)
+	if err != nil {
+		mod.LogFatal("Unable to start discordbee! Error: ", err)
+	}
+	mod.discord = d
+	mod.Logf("Done!")
+
+	mod.discord.AddHandler(mod.isReady)
+	mod.discord.AddHandler(mod.onSend)
+
+	err = mod.discord.Open()
+
+	if err != nil {
+		mod.LogFatal("Error opening websocket: ", err)
+	}
+
+	defer mod.discord.Close()
+
+	select {
+	case <-mod.SigChan:
+		return
+	}
+}
+
+func (mod *DiscordBee) isReady(s *discordgo.Session, event *discordgo.Ready) {
+	mod.Logf("Is ready")
+	s.UpdateStatus(0, "with bees")
```

Comment:
I'm a bit confused by this. Are we required to set some kind of status to make this work, or was this just part of a test?

Reply:
Updating the status is just a decorative thing - it should actually probably be a configuration option!

Reply:
I see! Maybe we should expose it as an action then? `set_status` or such.

Reply:
Ah, that's not a bad idea! Adding that to the list of things to do :)